pkg/cvo/availableupdates.go

```diff
@@ -164,8 +171,44 @@ func checkForUpdate(clusterID, upstream, channel string, currentVersion semver.V
 	if err != nil {
 		return nil, err
 	}
+	transport, err := getHTTPTransportProxy()
+	if err != nil {
+		return nil, err
+	}
+
 	if len(upstream) == 0 {
 		return nil, fmt.Errorf("no upstream URL set for cluster version")
 	}
-	return cincinnati.NewClient(uuid).GetUpdates(upstream, channel, currentVersion)
+	return cincinnati.NewClient(uuid, transport).GetUpdates(upstream, channel, currentVersion)
+}
+
+func getHTTPTransportProxy() (*http.Transport, error) {
+	transport := http.Transport{}
+	config, err := rest.InClusterConfig()
```

Comment:
You cannot create a new config here, please reuse the one in the operator struct already.

Or add proxy type client /listers to the operator struct and use here...

That would help in integration tests